component.js

```diff
@@ -95,19 +95,21 @@ module.exports.withDefaults = factory;
 function factory (options) {
   var debug;
   options = options || {};
-  var _shouldComponentUpdate = options.shouldComponentUpdate;
+  var _componentWillMount = options.componentWillMount ||
+                            componentWillMount;
+  var _shouldComponentUpdate = options.shouldComponentUpdate ||
+                               shouldComponentUpdate.withDefaults(options);
+  var _componentWillReceiveProps = options.componentWillReceiveProps ||
```

Comment:
http://tinyurl.com/klnmw3t shows that both life cycle mixins are called. This is with React v0.12.2.

Yeah, I think this isn't something you'd want to / need to override. As it's just more default behaviour / core stuff. I can't imagine anyone wanting to override it. I might be wrong of course.

> I don't really understand what you mean here. If you referring to order of lines above, it does not.

I was thinking of the order the mixins are called. If we need to have the internal mixins be invoked before any "external" mixins.